.github/workflows/version.js

```diff
@@ -0,0 +1,40 @@
+const { appendFileSync, readFileSync, writeFileSync } = require("fs");
+const { EOL } = require("os");
+const { GITHUB_OUTPUT, GITHUB_REF_NAME } = process.env;
+
+if (GITHUB_REF_NAME) {
+  const version = GITHUB_REF_NAME.startsWith("v") ? GITHUB_REF_NAME.slice(1) : GITHUB_REF_NAME;
+  addOutput("version", version);
+  addOutput("archive_name", `CLEO_${GITHUB_REF_NAME}.zip`);
+
+  // update cleo.h to replace version
+  const cleoH = readFileSync("cleo_sdk/cleo.h", { encoding: "utf-8" });
+  const newCleoH = cleoH.replace(/#define CLEO_VERSION_STR .*/, `#define CLEO_VERSION_STR "${version}"`);
```

Comment:
Defines
`#define CLEO_VERSION_MAIN	5
#define CLEO_VERSION_MAJOR	0
#define CLEO_VERSION_MINOR	0`
should be used instead, as these are used also for for example plugin version compatibility checks.

Reply:
I'll update both. CLEO_VERSION can only work with numbers, and I really want to preserve -alpha/beta suffix in the CLEO_VERSION_STR

Reply:
Hmm, in this case CLEO_VERSION_STR should probably only be modified in alpha builds.

Reply:
I don't think it matters. Release tag has the same format (main.major.minor) but with optional suffix that we only use in CLEO_VERSION_STR.

Reply:
If suffix is not there, we get the same result anyway

Reply:
It doubles amount of places where version info is declared. With macros it was impossible to produce incoherent version info.

Reply:
Version info is defined in code and in a github tag. Tag overwrites code. What is the issue?

Reply:
Is code in repo overwriten, or just the files inside release archive?